k6: 0.38.3 -> 0.39.0

bryanasdev000 opened the pull request:
###### Description of changes

- bump version
- drop explict docheck
- add shell completion for bash,fish and zsh

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Mic92 commented on pkgs/development/tools/k6/default.nix:
```diff
@@ -1,26 +1,34 @@
     $out/bin/k6 version | grep ${version} > /dev/null
   '';
 
+  postInstall = ''
```
To not break cross compilation:

```suggestion
  postInstall = lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
```

bryanasdev000 replied:
Welp, can you explain/reference a bit this part?

I have a postInstall to install shell files in a bunch of other Golang based projects.

smancill replied:
I guess it is because you are using `$out/bin/k6` in the hook.

Mic92 replied:
Yes. If you cross compile i.e. hostPlatform is aarch64 and buildPlatform is x86_64, than you cannot run your $out/bin/k6 binary compiled for aarch64 on x86_64.

bryanasdev000 replied:
Thanks @smancill @Mic92, will keep in mind to refactor my packages.

bryanasdev000 committed:
k6: 0.38.3 -> 0.39.0

- drop explict docheck

- add shell completion for bash,fish and zsh

- preserve cross compilation for shell files

smancill commented:
Result of `nixpkgs-review pr 182772` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>k6</li>
  </ul>
</details>

superherointj commented:
Result of `nixpkgs-review pr 182772` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>k6</li>
  </ul>
</details>
